### PR TITLE
helper-cli: Simplify parsing comma separated command line options

### PR DESCRIPTION
**Pull request by fviernau**
Signed-off-by: Frank Viernau <frank.viernau@here.com>

### CODE REVIEW

File: helper-cli/src/main/kotlin/commands/MergeRepositoryConfigurationsCommand.kt [hunk 1]
```diff
@@ -40,7 +40,7 @@ internal class MergeRepositoryConfigurationsCommand : CliktCommand(
     private val inputRepositoryConfigurationFiles by option(
         "--input-repository-configuration-files", "-i",
         help = "A comma separated list of the repository configuration files to be merged."
-    ).convert { it.split(",").toList().map { File(it.expandTilde()) } }
+    ).convert { csv -> csv.split(",").toList().map { File(it.expandTilde()) } }
```
Comment (sschuberth) [inline helper-cli/src/main/kotlin/commands/MergeRepositoryConfigurationsCommand.kt:43]: How about fixing this by applying a simplification like `.convert { File(it.expandTilde()) }.split(",").required()`?

Comment (sschuberth) [reply helper-cli/src/main/kotlin/commands/MergeRepositoryConfigurationsCommand.kt:43]: BTW, if you feel like it, similar simplification could be done in `ListLicensesCommand` and `ListPackagesCommand` where you also call `split()` "manually".

Comment (fviernau) [reply helper-cli/src/main/kotlin/commands/MergeRepositoryConfigurationsCommand.kt]: Good idea! Done.

File: helper-cli/src/main/kotlin/commands/ListLicensesCommand.kt [hunk 2]
```diff
@@ -93,8 +93,7 @@ internal class ListLicensesCommand : CliktCommand(
     private val ignoreExcludedRuleIds by option(
         "--ignore-excluded-rule-ids",
         help = "A comma separated list of rule names for which --omit-excluded should not have any effect."
-    ).convert { it.split(",").toList() }
-        .default(emptyList())
+    ).convert { File(it.expandTilde()) }.split(",").required()
```
Comment (sschuberth) [inline helper-cli/src/main/kotlin/commands/ListLicensesCommand.kt:96]: This should be `.split(",").default(emptyList())`.

File: helper-cli/src/main/kotlin/commands/ListPackagesCommand.kt [hunk 3]
```diff
@@ -44,8 +47,7 @@ class ListPackagesCommand : CliktCommand(
     private val matchDetectedLicenses by option(
         "--match-detected-licenses",
         help = "Omit all packages not matching all licenses given by this comma separated list of license identifiers."
-    ).convert { it.split(",").toList() }
-        .default(emptyList())
+    ).convert { File(it.expandTilde()) }.split(",").default(emptyList())
```
Comment (sschuberth) [inline helper-cli/src/main/kotlin/commands/ListPackagesCommand.kt:50]: This should be `.split(",").default(emptyList())`.

Comment (fviernau) [reply helper-cli/src/main/kotlin/commands/ListPackagesCommand.kt:50]: Done.

File: helper-cli/src/main/kotlin/commands/ListLicensesCommand.kt (hunk 2 above)
Comment (fviernau) [reply helper-cli/src/main/kotlin/commands/ListLicensesCommand.kt]: Done.

File: helper-cli/src/main/kotlin/commands/ListLicensesCommand.kt [hunk 4]
```diff
@@ -93,8 +93,7 @@ internal class ListLicensesCommand : CliktCommand(
     private val ignoreExcludedRuleIds by option(
         "--ignore-excluded-rule-ids",
         help = "A comma separated list of rule names for which --omit-excluded should not have any effect."
-    ).convert { it.split(",").toList() }
-        .default(emptyList())
+    ).convert { File(it.expandTilde()) }.split(",").default(emptyList())
```
Comment (sschuberth) [inline helper-cli/src/main/kotlin/commands/ListLicensesCommand.kt:96]: This is still wrong, `.convert { File(it.expandTilde()) }` should go away as this no `File` option.

File: helper-cli/src/main/kotlin/commands/ListPackagesCommand.kt (hunk 3 above)
Comment (sschuberth) [reply helper-cli/src/main/kotlin/commands/ListPackagesCommand.kt:50]: No, see above, this should *only* be `.split(",").default(emptyList())`.

File: helper-cli/src/main/kotlin/commands/ListLicensesCommand.kt (hunk 4 above)
Comment (fviernau) [reply helper-cli/src/main/kotlin/commands/ListLicensesCommand.kt]: Sorry.

File: helper-cli/src/main/kotlin/commands/ListPackagesCommand.kt (hunk 3 above)
Comment (fviernau) [reply helper-cli/src/main/kotlin/commands/ListPackagesCommand.kt]: Sorry.

File: helper-cli/src/main/kotlin/commands/MergeRepositoryConfigurationsCommand.kt (hunk 1 above)
Comment (fviernau) [reply helper-cli/src/main/kotlin/commands/MergeRepositoryConfigurationsCommand.kt:43]: Maybe I should make a break, but is it just me or did the original proposal not work due to reversed order of split and convert?

Comment (sschuberth) [reply helper-cli/src/main/kotlin/commands/MergeRepositoryConfigurationsCommand.kt:43]: I didn't test it, but IMO it should work, as we're doing something similar e.g. at https://github.com/oss-review-toolkit/ort/blob/90d12ee0c008967581fbca2a41e15a8a9744427c/cli/src/main/kotlin/commands/AnalyzerCommand.kt#L76

I.e. with *clikt* the order is "reversed" in than [you first specify the conversion, and then the split operation](https://ajalt.github.io/clikt/api/clikt/com.github.ajalt.clikt.parameters.options/split/). This seems counter-intuitive at first, but the `split()` doesn't really split immediately, but records a transformation to be performed later.

**Pull request merged**